### DISCUSSION TITLE
Add Sofware part to store Measurement Data of Energy Meter directly into MariaDB Database

### DISCUSSION THREAD

**Discussion by hasenradball** [arendst/Tasmota#18447] [Ideas]
Hi,

I would suggest to add a SW part to store the measured Data of the Energy Meter directly into a MariaDB database.
How is this idea?

**Comment by sfromis** [comment]
That type of specific integration depends on who really feels a need to do the work to create a feature, instead of a simple setup using Node-Red (or whatever) to bridge the communication. Suggesting such a feature works best if you code and provide it :grinning:

**Comment by Noschvie** [comment]
Usually a time series database (TSDB) like InfluxDB or Graphite is used to store measurements and displaying the results in Grafana.

**Reply by sfromis** [reply]
And Tasmota already has an integration for InfluxDB

**Reply by jkwim** [reply]
Could you point me to any doc which explains the integration from Tasmota side please?

**Reply by sfromis** [reply]
https://tasmota.github.io/docs/Commands/#influxdb
https://github.com/arendst/Tasmota/blob/development/tasmota/my_user_config.h#L403-L412

InfluxDB support is included in ESP32 binaries, but if you use an ESP8266-based device, you need a custom build including `#define USE_INFLUXDB` - the other settings shown in `my_user_config.h` are not needed when setting values in `user_config_override.h`, as they are also available via commands.

**Reply by jkwim** [reply]
Thanks for the quick reply. Will look in the links given.

**Comment by hasenradball** [comment]
Ok I already hav ESP s running which sends data to MySQL Database.
Therefore an Example is available.
InfluxDB would be good for timeseries data thats right.
Does anyone has an hint how to install influxDB on a pi.
I have already all my Data in MySQL.

**Reply by barbudor** [reply]
With docker it takes 2 minutes to install. And if you don't like it 10 seconds to get rid of it
Confguration however takes a little longuer to understand the concepts. It is not at all like SQL.

**Reply by hasenradball** [reply]
I never used a docker. Is there some good introduction?

**Reply by Noschvie** [reply]
@hasenradball Have a look to [IOTStack](https://sensorsiot.github.io/IOTstack/)

**Reply by barbudor** [reply]
I hate that stuff which simple things behind a complex front end
Docker is simple to use when you passed the first learning curve and you can control exactly
With iotstack, you don't understand what's your doing

**Comment by Jason2866** [comment]
ChatGPT will help with Docker questions

**Reply by barbudor** [reply]
Or it will lie
Who knows

Installing docker : https://docs.docker.com/engine/install/debian/

Installing InfluxDb on docker : https://docs.influxdata.com/influxdb/v2.7/install/?t=Docker

**Reply by hasenradball** [reply]
Thanks i will check it out 😉

**Comment by hasenradball** [comment]
@sfromis, Hi guys,

if I want to use the smart meter language scripting, do I need to compile the software by myself?

**Comment by Jason2866** [comment]
You can use the scripting build from https://tasmota.github.io/install/ in section unofficial
It has SML included

**Reply by hasenradball** [reply]
@Jason2866 When using this unofficial version can I also send the smart meter values via MQTT?

**Reply by sfromis** [reply]
Yes, "SML included" is about the smart meter feature.

**Reply by hasenradball** [reply]
Is there a small example how to read smartmeter data via MQTT.

One point is unclear:
how to set the host correctly?

a) ip like 192.168.xxx.xxx
b) ip like http://192.168.xxx.xxx
c) host name like "mosq_server"

Maybee a simple example would be great.

**Reply by sfromis** [reply]
a or c. It is no URL, just a host.

**Comment by hasenradball** [comment]
I got my Iskra MT681 to work, but got no values for the actual Power for the L1, L2,L3.
Does anyone know such a Problem?

**Reply by Noschvie** [reply]
Is your meter enabled with a PIN ?

**Reply by Noschvie** [reply]
Have a look to [this](https://forum.iobroker.net/topic/60654/gel%C3%B6st-iskra-mt681-mit-tasmota-auslesen/3)

**Reply by hasenradball** [reply]
Yea meter i enabled by Pin.

**Comment by hasenradball** [comment]
Sorry guys can anyone help me?
I have the following script and I wand to make it work to read two sensors.
```
>D
>B
=>sensor53 r
; Lichtstrom über MT681
>M 1
+1,4,s,0,9600,MT681
1,77070100010800ff@1000,1.8.0:,kWh,Total_in,3
1,77070100100700ff@1,Leistung:,W,Power_cur,0
;1,77070100240700ff@1,Leistung P1:,W,Power_p1,0
;1,77070100380700ff@1,Leistung P2:,W,Power_p2,0
;1,770701004c0700ff@1,Leistung P3:,W,Power_p3,0
1,77070100020800ff@1000,2.8.0:,kWh,Total_out,3
1,77070100000009ff@#,Service ID:,,Meter_id,0|
#
```
I tried to copy and paste the Sensor 1 and make a sensor 2 but doesn't work for me.

for what stands the `#` at the end?

**Comment by hasenradball** [comment]
Can anyone explain what does the `| #` at the end mean?

```
1,77070100000009ff@#,Service ID:,,Meter_id,0|
#
```
